x-pack/plugins/ml/server/models/data_frame_analytics/analytics_manager.ts

```diff
@@ -0,0 +1,443 @@
+  isJobDataLinkReturnType,
+  isTransformLinkReturnType,
+  MapElements,
+  NextLinkReturnType,
+} from './types';
+import type { MlClient } from '../../lib/ml_client';
+
+export class AnalyticsManager {
+  private _client: IScopedClusterClient['asInternalUser'];
+  private _mlClient: MlClient;
+  public _inferenceModels: any; // TODO: update types
+
+  constructor(mlClient: MlClient, client: IScopedClusterClient['asInternalUser']) {
+    this._client = client;
+    this._mlClient = mlClient;
+    this._inferenceModels = [];
+  }
+
+  public set inferenceModels(models: any) {
+    this._inferenceModels = models;
+  }
+
+  public get inferenceModels(): any {
+    return this._inferenceModels;
+  }
+
+  async setInferenceModels() {
+    try {
+      const models = await this.getAnalyticsModels();
+      this.inferenceModels = models;
+    } catch (error) {
+      // TODO: bubble up this error?
+      // eslint-disable-next-line
+      console.error('Unable to fetch inference models', error);
+    }
+  }
+
+  private isDuplicateElement(analyticsId: string, elements: any[]): boolean {
+    let isDuplicate = false;
+    elements.forEach((elem: any) => {
+      if (elem.data.label === analyticsId && elem.data.type === JOB_MAP_NODE_TYPES.ANALYTICS) {
+        isDuplicate = true;
+      }
+    });
+    return isDuplicate;
+  }
+  // @ts-ignore // TODO: is this needed?
+  private async getAnalyticsModelData(modelId: string) {
+    const resp = await this._mlClient.getTrainedModels({
+      model_id: modelId,
+    });
+    const modelData = resp?.body?.trained_model_configs[0];
+    return modelData;
+  }
+
+  private async getAnalyticsModels() {
+    const resp = await this._mlClient.getTrainedModels();
+    const models = resp?.body?.trained_model_configs;
+    return models;
+  }
+
+  private async getAnalyticsJobData(analyticsId: string) {
+    const resp = await this._mlClient.getDataFrameAnalytics({
+      id: analyticsId,
+    });
+    const jobData = resp?.body?.data_frame_analytics[0];
+    return jobData;
+  }
+
+  private async getIndexData(index: string) {
+    const indexData = await this._client.indices.get({
+      index,
+    });
+
+    return indexData?.body;
+  }
+
+  private async getTransformData(transformId: string) {
+    const transform = await this._client.transform.getTransform({
+      transform_id: transformId,
+    });
+    const transformData = transform?.body?.transforms[0];
+    return transformData;
+  }
+
+  private findJobModel(analyticsId: string): any {
+    return this.inferenceModels.find(
+      (model: any) => model.metadata?.analytics_config?.id === analyticsId
+    );
+  }
+
+  private async getNextLink({
+    id,
+    type,
+  }: {
+    id: string;
+    type: JobMapNodeTypes;
+  }): Promise<NextLinkReturnType> {
+    try {
+      if (type === JOB_MAP_NODE_TYPES.INDEX_PATTERN) {
+        // fetch index data
+        const indexData = await this.getIndexData(id);
+        const meta = indexData[id].mappings._meta;
```

Comment:
I think this is what is causing me to see this error on my outlier job on a gallery transform.

![image](https://user-images.githubusercontent.com/7405507/98359404-151a0080-2020-11eb-9d47-559c0005e20b.png)


The transform was created against the `gallery-*` index pattern, but it seems there isn't a key in `indexData` for `gallery-*` so `indexData[id]` is `undefined` here. Added in some debug and these are the keys for my `indexData` object:

```
getNextLink indexData keys: [ 'gallery-2019-08',
  'gallery-2019-09',
  'gallery-2019-10',
  'gallery-2019-11',
  'gallery-2019-12',
  'gallery-2020-01',
  'gallery-2020-02',
  'gallery-2020-03',
  'gallery-2020-04',
  'gallery-2020-05' ]
```

Reply:
Added to meta issue. In the case of a wildcard - it might be a good idea to simply return all the source index nodes and then let the user choose which path to follow from the UI. Thoughts, @peteharverson ?